modules/ROOT/pages/anypoint-mq/anypoint-mq-listener.adoc

```diff
@@ -88,6 +88,141 @@ If instead, an error occurs during the execution of the flow, the session is rec
+
+
+=== How does it work?
+
+The circuit breaker capability provided by the `anypoint-mq:subscriber` will be bound to the error handling mechanism that Mule provides out of the box, using the errors notification mechanism to keep count of errors related to our external service which we'll refer to occurences of this errors as "circuit failures". Any error can be bound to be considered a circuit failure, for example you can bound `HTTP:TIMEOUT`, `FTP:SERVICE_NOT_AVAILABLE`, or even a custom error from your application like `ORG:EXTERNAL_ERROR`.
+
+Every time a mule flow completes its execution with an error, the subscriber will check if the error is one of the `onErrorTypes` that indicate an external service error, counting consecutive occurences until the `errorsThreshold` is reached. When the `errorsThreshold` is reached, _the circuit will be tripped_, effectively stopping the poll for new messages for a configurable `tripTimeout`. Messages will be consumed again on the next poll where the tripTimeout has evicted.
+By default, circuit breaking feature will be disabled.
+
+=== States Overview
+
+image::mq-subscriber-states-view.png[caption="Circuit Breaker States Overview"]
+
+
+- Closed
+
+    This is the starting state, where the subscriber will retrieve messages normally from MQ based on its configuration, effectively working as if the Circuit Breaker is not present.
+
+- Closed-Open Transition
+
+    When the amount of failures occurred one after the other during message processing, without any success in between the failures, reaches the `errorsThreshold` value, then the circuit breaker will trip and transition to an _open_ state.
+    At this point, messages that were already dispatched to the flow will complete its processing (either if the result is success or failure).
+    Messages kept locally, which are in-flight for the broker but haven't been dispatched yet, will be NACKed, effectively returning to the Queue for redelivery to other consumer.
+
+- Open
+
+    In this state, the subscriber won't attempt to retrieve messages, skipping the iterations silently until the `tripTimeout` is reached.
+
+- Half Open
+
+    Once the `tripTimeout` has elapsed, the subscriber will go to a _half open_ state meaning that in the next poll for messages, it will retrieve __a single message_ from the service and use that message to check if the system has recovered before going back to the normal, _closed_, state.
+    When single message is successfully fetch, dispatched to the flow and its processing completes with a success, then the subscriber will go back to normal, immediately attempting to fetch more messages.
```

Comment:
slight correction to the first sentence: When **a** single message is successfully **fetched**